[BUG] Levant cannot handle system jobs

As a result of #16 and the merge that is associated Levant can not currently (0.0.3-dev) handle system jobs. This is because system jobs do not have a taskgroup count and the code does not handle this causing a panic:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x1735bcc]

goroutine 1 [running]:
github.com/jrasell/levant/levant.(*nomadClient).Deploy(0xc42015e020, 0xc4200a1b00, 0x0, 0xc42015e000, 0x0)
	/Users/rasellj/go/src/github.com/jrasell/levant/levant/deploy.go:66 +0xbc
github.com/jrasell/levant/command.(*DeployCommand).Run(0xc4202036e0, 0xc420010180, 0x3, 0x3, 0xc4200e3940)
	/Users/rasellj/go/src/github.com/jrasell/levant/command/deploy.go:109 +0x58b
github.com/jrasell/levant/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc4200b5a40, 0xc4200b5a40, 0x3, 0xc4200e39a0)
	/Users/rasellj/go/src/github.com/jrasell/levant/vendor/github.com/mitchellh/cli/cli.go:255 +0x1eb
main.RunCustom(0xc420010150, 0x4, 0x4, 0xc420203590, 0xc4202034d0)
	/Users/rasellj/go/src/github.com/jrasell/levant/main.go:49 +0x433
main.Run(0xc420010150, 0x4, 0x4, 0x173da88)
	/Users/rasellj/go/src/github.com/jrasell/levant/main.go:17 +0x56
main.main()
	/Users/rasellj/go/src/github.com/jrasell/levant/main.go:11 +0x63
```

--- a/levant/deploy.go
+++ b/levant/deploy.go
@@ -60,14 +60,18 @@ func (c *nomadClient) Deploy(job *nomad.Job, autoPromote int, forceCount bool) (
 		}
 	}
 
-	// Check that the job has at least 1 TaskGroup with count > 0 (GH-16)
-	TGCount := 0
-	for _, group := range job.TaskGroups {
-		TGCount += *group.Count
-	}
-	if TGCount == 0 {
-		logging.Error("levant/deploy: all TaskGroups have a count of 0, nothing to do")
-		return
+	// Check that the job has at least 1 TaskGroup with count > 0 (GH-16) if the
+	// job is not a System job. Systems jobs do not define counts so cannot be
+	// checked.
+	if *job.Type != nomadStructs.JobTypeSystem {
+		tgCount := 0
+		for _, group := range job.TaskGroups {
+			tgCount += *group.Count
+		}
+		if tgCount == 0 {
+			logging.Error("levant/deploy: all TaskGroups have a count of 0, nothing to do")
+			return
+		}
 	}
 
 	logging.Info("levant/deploy: triggering a deployment of job %s", *job.Name)

